### PR TITLE
GH-50 Added smart remote repo settings

### LINKED ISSUE #50: Remote repository - Smart remote repository settings reset after terraform apply
**Describe the bug**
When configuring remote repository, **Smart Remote Repository** settings are reseted to default values even if those settings are not supported by Artifactory provider. ("Report Statistics" & "Sync Properties" & "Source Absence Detection" checkboxes are cleared, "List Remote Folder Items" stays checked)

**Requirements for and issue**
- [ ] Snippet
```
resource "artifactory_remote_repository" "repo-remote" {
  provider     = artifactory.myinstance
  key          = "nuget-repo-remote"
  package_type = "nuget"
  description  = "(local file cache)"
  url          = "https://artifactory.company.com/artifactory/api/nuget/nuget-location-a-local/"
  username     = var.art_login_cloud_account
  password     = var.art_login_cloud_password
  offline      = false

  allow_any_host_auth          = false
  blacked_out                  = false
  block_mismatching_mime_types = true
  bypass_head_requests         = true
  content_synchronisation {
    enabled = true
  }
  download_context_path                 = "Download"
  enable_cookie_management              = false
  enable_token_authentication           = false
  feed_context_path                     = null
  fetch_jars_eagerly                    = false
  fetch_sources_eagerly                 = false
  force_nuget_authentication            = false
  handle_releases                       = true
  handle_snapshots                      = true
  hard_fail                             = false
  includes_pattern                      = "**/*"
  max_unique_snapshots                  = 0
  missed_cache_period_seconds           = 900
  notes                                 = "abcd note"
  property_sets                         = ["artifactory"]
  remote_repo_checksum_policy_type      = "generate-if-absent"
  repo_layout_ref                       = "nuget-abcd-layout"
  retrieval_cache_period_seconds        = 1800
  share_configuration                   = false
  socket_timeout_millis                 = 15000
  store_artifacts_locally               = true
  suppress_pom_consistency_checks       = true
  synchronize_properties                = false
  unused_artifacts_cleanup_period_hours = 168
  v3_feed_url                           = "https://api.nuget.org/v3/index.json"
}
```
- [ ] Version of artifactory: 7.13.4
- [ ] Version of terraform: 0.13.6
- [ ] Version of provider: 2.2.7

**Expected behavior**
When "Smart Remote Repository" are set up using WebUI, terraform apply command does not affect them when it is not supported by provider. Or better way, include these settings in provider.

### PR DESCRIPTION
**Pull request by maheshjfrog**
Fixes #50

### CODE REVIEW

File: docs/resources/artifactory_remote_cargo_repository.md [hunk 1]
```diff
@@ -27,4 +27,9 @@ Arguments have a one to one mapping with the [JFrog API](https://www.jfrog.com/c
 * `key` - (Required) The repository identifier. Must be unique system-wide
 * `anonymous_access` - (Required) - Cargo client does not send credentials when performing download and search for crates. Enable this to allow anonymous access to these resources (only), note that this will override the security anonymous access option.
 * `git_registry_url` - (Optional) - This is the index url, expected to be a git repository. for remote artifactory use "arturl/git/repokey.git"
+* `content_synchronisation` - (Optional) Reference [JFROG Smart Remote Repositories](https://www.jfrog.com/confluence/display/JFROG/Smart+Remote+Repositories)
```
Comment (alexhung) [inline docs/resources/artifactory_remote_cargo_repository.md:30]: URL is missing?

File: pkg/artifactory/repositories.go [hunk 2]
```diff
@@ -49,7 +49,22 @@ func (bp LocalRepositoryBaseParams) Id() string {
 }
 
 type ContentSynchronisation struct {
-	Enabled bool `hcl:"enabled" json:"enabled,omitempty"`
+	Enabled    bool                             `hcl:"enabled" json:"enabled"`
```
Comment (alexhung) [inline pkg/artifactory/repositories.go:52]: No need for `hcl` tag

File: pkg/artifactory/resource_artifactory_remote_repository.go [hunk 3]
```diff
@@ -319,12 +318,31 @@ var legacyRemoteSchema = map[string]*schema.Schema{
 		Type:     schema.TypeList,
 		Optional: true,
 		Computed: true,
-		MaxItems: 1,
```
Comment (alexhung) [inline pkg/artifactory/resource_artifactory_remote_repository.go:322]: We don't need to constraint to one item any more? How would that work if user defines more than one in HCL?

**Review by alexhung — COMMENTED** [review]
Look pretty good. A few minor questions

File: docs/resources/artifactory_remote_cargo_repository.md (hunk 1 above)
Comment (maheshjfrog) [reply docs/resources/artifactory_remote_cargo_repository.md:30]: URL is set to https://www.jfrog.com/confluence/display/JFROG/Smart+Remote+Repositories ... right?

File: pkg/artifactory/resource_artifactory_remote_repository.go (hunk 3 above)
Comment (maheshjfrog) [reply pkg/artifactory/resource_artifactory_remote_repository.go:322]: Will limit to 1. 
Assumed that limit was for attributes under 'content_synchronisation' but it was for 'content_synchronisation' block itself.

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,6 +4,10 @@ IMPROVEMENTS:
 
 * Add missing documentations for Federated repo resources [GH-304]
 * Add additional repo types for Federated repo resources [GH-304]
+* Added following smart remote repo attributes for remote repository resources [GH-50].
```
Comment (alexhung) [inline CHANGELOG.md:7]: Please create a new section for 2.14.0

Comment (alexhung) [reply CHANGELOG.md:7]: I don't think TF actually replaces `(Unreleased)` with actual dates. So please also add today's date.

**Review by alexhung — CHANGES_REQUESTED** [review]
Apart changes need in CHANGELOG, this is ready to merge.

File: CHANGELOG.md (hunk 4 above)
Comment (alexhung) [reply CHANGELOG.md:7]: The GH reference is for the PR, not the original issue.

**Review by alexhung — CHANGES_REQUESTED** [review]

File: pkg/artifactory/resource_artifactory_remote_repository.go [hunk 5]
```diff
@@ -323,8 +322,28 @@ var legacyRemoteSchema = map[string]*schema.Schema{
 		Elem: &schema.Resource{
 			Schema: map[string]*schema.Schema{
 				"enabled": {
-					Type:     schema.TypeBool,
-					Optional: true,
+					Type:        schema.TypeBool,
```
[the remark this answers is not in the document]
Comment (maheshjfrog) [reply pkg/artifactory/resource_artifactory_remote_repository.go:325]: Ok. Removed this enhancement for old remote repo schema.

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,10 +1,14 @@
-## 2.13.1 (Unreleased)
+## 2.14.0 (Feb, 2, 2022)
 
 IMPROVEMENTS:
 
 * Add missing documentations for Federated repo resources [GH-304]
 * Add additional repo types for Federated repo resources [GH-304]
-
+* Added following smart remote repo attributes for remote repository resources [GH-305].
```
[the remark this answers is not in the document]
Comment (maheshjfrog) [reply CHANGELOG.md]: Created new section for 2.14.0 for this PR.

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,10 +1,18 @@
+## 2.14.0 (Feb, 3, 2022)
+FEATURES:
+
+* Added following smart remote repo attributes for npm, cargo, docker and helm remote repository resources [GH-305].
+  * "statistics_enabled"
+  * "properties_enabled"
+  * "source_origin_absence_detection"
+
 ## 2.13.1 (Unreleased)
```
Comment (alexhung) [inline CHANGELOG.md:9]: Can you update this with the release date? Thanks!

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,10 +1,18 @@
+## 2.14.0 (Feb, 3, 2022)
+FEATURES:
```
Comment (alexhung) [inline CHANGELOG.md:2]: We need an empty line between the version/date and the category, per TF guideline.

File: CHANGELOG.md (hunk 7 above)
Comment (maheshjfrog) [reply CHANGELOG.md]: done

File: CHANGELOG.md (hunk 8 above)
Comment (maheshjfrog) [reply CHANGELOG.md:3]: done

**Review by alexhung — APPROVED** [review]

File: pkg/artifactory/repositories.go [hunk 9]
```diff
@@ -719,8 +754,20 @@ func unpackBaseRemoteRepo(s *schema.ResourceData, packageType string) RemoteRepo
 	if v, ok := d.GetOk("content_synchronisation"); ok {
 		contentSynchronisationConfig := v.([]interface{})[0].(map[string]interface{})
 		enabled := contentSynchronisationConfig["enabled"].(bool)
+		statisticsEnabled := contentSynchronisationConfig["statistics_enabled"].(bool)
+		propertiesEnabled := contentSynchronisationConfig["properties_enabled"].(bool)
+		sourceOriginAbsenceDetection := contentSynchronisationConfig["source_origin_absence_detection"].(bool)
 		repo.ContentSynchronisation = &ContentSynchronisation{
 			Enabled: enabled,
+			Statistics: ContentSynchronisationStatistics{
```
Comment (chb0github) [inline pkg/artifactory/repositories.go:762]: So, I was meaning to ask: `ContentSynchronization` is a pointer here and I don't see the need. I won't ask that you change it, but when you come upon code like this you're aware and try to actually eliminate pointers

**Review by chb0github — APPROVED** [review]

**Pull request merged**